### PR TITLE
archive.Writer: fix staleDuration

### PR DESCRIPTION
**Pull request by mattnibs**
Previously archive.Writer was setting the flusher interval to whatever the
flush timeout was. This didn't make sense and was also causing import test
flakiness. Set the flush interval to an arbitrary one second and make
staleDuration configurable.

### CODE REVIEW

File: ppl/archive/import.go [hunk 1]
```diff
@@ -69,22 +65,22 @@ type Writer struct {
...
+		errgroup: g,
+		writers:  make(map[tsDir]*tsDirWriter),
 	}
 }
 
-// SetFlushTimeout sets the interval at which the Writer will wait to flush
-// stale records to disk. This must be called before writing any records.
-func (w *Writer) SetFlushTimeout(timeout time.Duration) {
-	w.flushTimeout = timeout
+// SetStaleDuration sets the StaleThreshold for the writer which is the
```
Comment (nwt) [inline ppl/archive/import.go:79]:
```suggestion
// SetStaleDuration sets the stale duration for the writer which is the
```

File: ppl/archive/import.go [hunk 2]
```diff
@@ -69,22 +65,22 @@ type Writer struct {
...
+// duration since a tsdir has last since received new data will  flushed to
```
Comment (alfred-landrum) [inline ppl/archive/import.go:80]: Reword comment.

Comment (nwt) [inline ppl/archive/import.go:80]:
```suggestion
// delay after the last write to a tsdir until it is flushed to
```

**Review by nwt — APPROVED** [review]

File: ppl/archive/import_test.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ func TestImportFlushTimeout(t *testing.T) {
 	})
 }
 
-func testImportFlushTimeout(t *testing.T, timeout time.Duration, expected uint64) {
+func testImportFlushTimeout(t *testing.T, stale time.Duration, expected uint64) {
```
Comment (nwt) [inline ppl/archive/import_test.go:26]:
```suggestion
func testImportStaleDuration(t *testing.T, stale time.Duration, expected uint64) {
```

File: ppl/archive/import.go (hunk 2 above)
Comment (mattnibs) [reply ppl/archive/import.go:80]: Yeah

Comment (mattnibs) [reply ppl/archive/import.go:80]: Thanks man, words on tough for me today.

**Pull request merged**